<launch>
	<arg name="veh" default="$(env VEHICLE_NAME)"/>
	<arg name="ai_trafo_mode" default="cb" doc="'cb' for colo balance only; 'both' for color balance and linear trafo"/>
	<arg name="ai_interval" default="5" doc="interval with which the linear trafo gets updated. color balance is performed every second."/>
	<arg name="verbose" default="false"/>

	<!-- start basic args -->
	<include file="$(find duckietown_demos)/launch/master.launch">
		<arg name="veh" value="$(arg veh)"/>
		<arg name="visualization" default="true" />
		<!--arg name="verbose" value="$(arg verbose)" /-->
    	<arg name="/camera/raw" default="true" />
		<arg name="param_file_name" default="default" />
		<arg name="line_detector_param_file_name" default="default" />
		<arg name="anti_instagram" default="true" />
		<arg name="ai_trafo_mode" value="$(arg ai_trafo_mode)"/>
		<arg name="ai_interval" value="$(arg ai_interval)"/>
		<arg name="/lane_following/stop_line_filter" default="false" /> <!-- will not every allow a transition into COORDINATION mode -->
		<arg name="vehicle_avoidance" default="false"/>
		<arg name="/lane_following/lane_controller" default="false"/>
	</include>

	<group ns="$(arg veh)">
		<remap from="pp_lane_controller/car_cmd" to="joy_mapper_node/car_cmd"/>
		<remap from="pp_lane_controller/corrected_image/compressed" to="anti_instagram_node/corrected_image/compressed"/>
		<remap from="pp_lane_controller/segment_list" to="line_detector_node/segment_list"/>
		<remap from="pp_lane_controller/lane_pose" to="lane_filter_node/lane_pose"/>
		<remap from="pp_lane_controller/lineseglist_out" to="ground_projection/lineseglist_out"/>
		<remap from="pp_lane_controller/seglist_filtered" to="lane_filter_node/seglist_filtered"/>
		<node pkg="pure_pursuit_lf" type="pp_lane_controller.py" name="pp_lane_controller" output="screen" required="true">
			<param name="verbose" value="$(arg verbose)"/>
		</node>
	</group>
</launch>
